### PR TITLE
Fixes to allow building with bundled llvm 17 with gpus

### PR DESCRIPTION
**Pull request by jabraham17**
This is a handful of fixes to allow building the compiler with CHPL_LLVM=bundled with llvm 17 and CHPL_GPU=nvidia. This also makes CHPL_LLVM=bundled and CHPL_GPU=amd together a config error, CHPL_GPU=amd a system llvm from rocm.

This PR also adjusts the configs for nightly rocm tests to use CHPL_LLVM=system

Tested with #24290 and should be merged with that PR

Testing
- [x] paratest without comm 
- [x] paratest with comm
- [x] `st test/gpu` without comm and CHPL_GPU=nvidia
- [x] `st test/gpu` with comm and CHPL_GPU=nvidia
- [x] Check that building the compiler with CHPL_LLVM=bundled and CHPL_GPU=amd is a config error

[Reviewed by @e-kayrakli]

### CODE REVIEW

File: compiler/llvm/clangUtil.cpp [hunk 1]
```diff
@@ -2754,6 +2754,7 @@ static bool isTargetCpuValid(const char* targetCpu) {
 static std::string generateClangGpuLangArgs() {
   std::string args = "";
   if (isFullGpuCodegen()) {
+    args += "-D__STRICT_ANSI__=1 ";
```
Comment (e-kayrakli) [inline compiler/llvm/clangUtil.cpp:2757]: Should we do this only for nvidia just to be on the more cautious side?

File: runtime/src/gpu/nvidia/Makefile.share [hunk 2]
```diff
@@ -28,7 +28,7 @@ GPU_OBJS = $(addprefix $(GPU_OBJDIR)/,$(addsuffix .o,$(basename $(GPU_SRCS))))
 # a comma-separated list.
 RUNTIME_CXXFLAGS += -x cuda -Wno-unknown-cuda-version \
                     -Xclang -fcuda-allow-variadic-functions \
-                    --offload-arch=$(CHPL_MAKE_GPU_ARCH)
+                    --offload-arch=$(CHPL_MAKE_GPU_ARCH) -D__STRICT_ANSI__=1
```
Comment (e-kayrakli) [inline runtime/src/gpu/nvidia/Makefile.share:31]: Was https://github.com/chapel-lang/chapel/pull/24303/files#diff-32227b333833b287cbe3c32f2ac41d8689fb69236c82f321fa039155c4cd5d09R72 not enough? It is not really a problem, but we might be adding this twice for `gpu-nvidia-reduce.cc`.

File: test/gpu/interop/cuBLAS/level1/test_cublas1.suppressif [hunk 3]
```diff
@@ -0,0 +1 @@
+CHPL_LOCALE_MODEL==gpu
```
Comment (e-kayrakli) [inline test/gpu/interop/cuBLAS/level1/test_cublas1.suppressif:1]: We can drop these suppressifs.

File: util/chplenv/compile_link_args_utils.py [hunk 4]
```diff
@@ -94,6 +96,9 @@ def get_runtime_link_args(runtime_subdir):
         # and add cuda libraries
         gpu_type = chpl_gpu.get()
         sdk_path = chpl_gpu.get_sdk_path(gpu_type)
+
+        # system.append("-std=c++17")
```
Comment (e-kayrakli) [inline util/chplenv/compile_link_args_utils.py:100]: drop

**Review by e-kayrakli — APPROVED** [review]

File: runtime/src/gpu/nvidia/Makefile.share (hunk 2 above)
Comment (jabraham17) [reply runtime/src/gpu/nvidia/Makefile.share:31]: yes you are right, this is extra with the changes in `compile_link_args_utils.py`. Removing from here.

**Pull request merged**